### PR TITLE
[RichDEM] Differentiate on Julia versions

### PR DESCRIPTION
**Pull request by barche**
This should fix issue https://github.com/JuliaInterop/libcxxwrap-julia/issues/115 and also attempts to add a few more platforms.

### CODE REVIEW

**Comment by barche** [comment]
Most failures are due of the style:

```
/workspace/destdir/lib/libgdal.so: undefined reference to `curl_easy_getinfo@CURL_4'
```

Any idea how to fix this, @giordano ? It seems in some cases there is a version mismatch with `libcurl`, which comes in through the `GDAL_jll` dependency, but it seems to correctly specify libcurl compat?

**Comment by giordano** [comment]
Sounds like you need to link to libcurl as well (but I don't why it wasn't a problem before).

**Comment by sunayana** [comment]
Earlier I was not building for all platforms supported by Julia, but then the generated binaries were not working as reported in the issue linked by @barche

**Comment by barche** [comment]
The apple failures are because OpenMP is not found, I think that can be solved by using `LLVMOpenMP_jll` on macOS as is done e.g. here:
https://github.com/barche/Yggdrasil/blob/3b5c42fac291891dba6636fc60d2cd157791a046/D/Darknet/common.jl#L68

Not sure about the others, but they are more obscure platforms, so they could be disabled if there is no easy fix.

**Comment by sunayana** [comment]
In the meanwhile I have also tested against the tarballs in artifacts generated by buildkite and can confirm that this PR solves my issues reported in https://github.com/JuliaInterop/libcxxwrap-julia/issues/115 so could I request this PR to be merged.

File: R/RichDEM/build_tarballs.jl [hunk 1]
```diff
@@ -17,24 +22,46 @@ sources = [
...
-    Platform("powerpc64le", "linux"; libc = "glibc"),
-    Platform("aarch64", "linux"; libc = "musl"),
-    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
-    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl"),
-]
-
+julia_versions = [v"1.6.3", v"1.8"]
+include("../../L/libjulia/common.jl")
+platforms = vcat(libjulia_platforms.(julia_versions)...)
+platformfilter(p) = (p.tags["arch"] != "armv6l" && p.tags["os"] ∉ ["macos", "freebsd"])
```
Comment (giordano) [inline R/RichDEM/build_tarballs.jl:62]: Avoid accessing tags, we have functions
```suggestion
platformfilter(p) = (arch(p) != "armv6l" && Sys.isbsd(p))
```

**Comment by barche** [comment]
Thanks for the update, I think this is OK. I'll click the "update branch" button and then it should be good.

**Comment by giordano** [comment]
> I'll click the "update branch" button and then it should be good.

Please don't, it's almost always totally useless.  I'm considering removing it again.

**Comment by sunayana** [comment]
Thank you very much @barche and @giordano

**Pull request merged**